src/from/b1505/BaseB1505.ts

```diff
@@ -94,8 +95,22 @@ export default class BaseB1505 {
           parsedMeta.SetupTitle,
         meta: parsedMeta,
       });
-      for (const calculation of this.calculations) {
-        calculation(analysis);
+
+      for (const [name, calculation] of this.calculations) {
+        const spectrum = analysis.getXYSpectrum({
+          xLabel: this.xLabel,
+          yLabel: this.yLabel,
+        });
+        if (spectrum) {
+          const result = JSON.stringify(calculation(spectrum));
+          if (result) {
+            if (spectrum.meta) {
+              spectrum.meta[name] = result;
+            } else {
+              spectrum.meta = { [name]: result };
```

Comment:
I don't understand the idea behind creating this object. You anyway replace the existing value.